plutus-core/plutus-ir/src/PlutusIR/Analysis/Usages.hs

```diff
@@ -26,15 +26,11 @@ addUsage n usages =
         old = Map.findWithDefault 0 u usages
     in Map.insert u (old+1) usages
 
--- | Check if @n@ is used at least once.
-isUsed :: (PLC.HasUnique n unique) => n -> Usages -> Bool
-isUsed n usages = Map.findWithDefault 0 (n ^. PLC.unique . coerced) usages > 0
+-- | Get the usage count of @n@.
+getUsageCount :: (PLC.HasUnique n unique) => n -> Usages -> Int
+getUsageCount n usages = Map.findWithDefault 0 (n ^. PLC.unique . coerced) usages
 
--- | Check if @n@ is used exactly once.
-isUsedOnce :: (PLC.HasUnique n unique) => n -> Usages -> Bool
-isUsedOnce n usages = Map.findWithDefault 0 (n ^. PLC.unique . coerced) usages == 1
-
--- | Get a set of all used @n@s.
+-- | Get a set of @n@s which are used at least once.
 allUsed :: Usages -> Set.Set PLC.Unique
 allUsed usages = Map.keysSet $ Map.filter (> 0) usages
```

Comment:
isn't the filter inside `allUsed` kind of redundant? We never store on Usages map anything that has less than 0 usages, right? But i guess better be exact/correct.

Reply:
Yes I guess, but it doesn't hurt :shrug:.